### PR TITLE
Add Single Product

### PR DESCRIPTION
**Pull request by derrickobedgiu1**
Extract from previous bundled PR

### CODE REVIEW

File: src/Message/SingleProductMessage.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private string $product_retailer_id;
+
+    private ?string $body;
+
+    private ?string $footer;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(string $to, int $catalog_id, string $product_retailer_id, ?string $body, ?string $footer, ?string $reply_to)
```
Comment (aalbarca) [inline src/Message/SingleProductMessage.php:23]: `$body, $footer, $reply_to` maybe should are defined with a default value? `null`?

File: tests/Integration/WhatsAppCloudApiTest.php [hunk 2]
```diff
@@ -301,6 +301,29 @@ public function test_send_reply_buttons()
         $this->assertEquals(false, $response->isError());
     }
 
+    public function test_send_single_product()
+    {
+        $this->markTestIncomplete(
+            'This test requires real catalog id and product sku id.'
```
Comment (aalbarca) [inline tests/Integration/WhatsAppCloudApiTest.php:307]: incompleted tests are not accepted

**Review by aalbarca — CHANGES_REQUESTED** [review]

File: tests/Integration/WhatsAppCloudApiTest.php [hunk 3]
```diff
@@ -301,6 +301,25 @@ public function test_send_reply_buttons()
         $this->assertEquals(false, $response->isError());
     }
 
+    public function test_send_single_product()
+    {
+        $catalog_id = '<catalog-id>';
```
Comment (aalbarca) [inline tests/Integration/WhatsAppCloudApiTest.php:306]: It is possible to create some object to retrieve a `catalog_id` and a `product_sku_id`?

Comment (derrickobedgiu1) [reply tests/Integration/WhatsAppCloudApiTest.php:306]: May be to get the catalog_id and product_sku_id from the sample test credentials file? If that's what you mean I can proceed to do that.

**Comment by aalbarca** [comment]
Hi @derrickobedgiu1 this branch is not mergeable due to conflicting files. You should rebase with main. Thanks!

**Comment by aalbarca** [comment]
Hi @derrickobedgiu1 did you close the PR? Why? Can I help you?

**Comment by aalbarca** [comment]
Hi @derrickobedgiu1  did you close the PR? Why? Can I help you?

**Comment by derrickobedgiu1** [comment]
Hello @aalbarca I tried rebasing and ended messing up as haven't done that before. Decided to just close both PRs.

**Comment by aalbarca** [comment]
See https://github.com/netflie/whatsapp-cloud-api/pull/209

**Pull request closed without merging**